### ISSUE TITLE
sslv3 alert handshake failure

### ISSUE THREAD

**Issue by bpriya8711** [open-quantum-safe/oqs-demos#15]
Hi 
After running the last command of Nginx I am getting the following output. Please help.

 C:\Users\openssl-OQS-OpenSSL_1_1_1-stable\nginx> docker exec oqs-nginx /opt/openssl/apps/openssl s_client -CAfile CA.crt -curves oqs_kem_default -connect localhost:4433
0:error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure:ssl/record/rec_layer_s3.c:1544:SSL alert number 40
CONNECTED(00000003)
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 7 bytes and written 9909 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---

**Comment by baentsch** [comment]
@bpriya8711 Thanks for the problem report. Indeed, a change in upstream code has led to the documentation to be out of sync with the code. Until this is fixed, here is a workaround:
Decide for a (set of) QSC KEM algorithm(s) to use (from the list [here](https://github.com/open-quantum-safe/liboqs#key-encapsulation-mechanisms)), then specify this in the file `nginx.conf`, e.g., as follows:
```
ssl_ecdh_curve oqs_kem_default:frodo976shake:frodo1344shake:kyber512:kyber768:kyber1024:newhope512cca:newhope1024cca:ntru_hps2048509:ntru_hps2048677:ntru_hrss701:lightsaber:saber:sidhp434:sidhp503:sikep751;
```

After building the docker image you can then select any of the KEX algorithms in this list in the command you want to execute above (`docker exec oqs-nginx /opt/openssl/apps/openssl s_client -CAfile CA.crt -curves oqs_kem_default -connect localhost:4433`) and it will work OK (i.e., a QSC-connection utilizing the algorithm you want to use is established).

**Comment by bpriya8711** [comment]
![image](https://user-images.githubusercontent.com/62378642/79963991-b340e180-84a7-11ea-8ee2-ee5b6b86fa1f.png)
Thanks for the prompt response, we are able to generate the certificate using the command.
" docker exec oqs-nginx /opt/openssl/apps/openssl s_client -CAfile CA.crt -curves oqs_kem_default -connect localhost:4433" after doing the changes in nginx.conf file.
When I tried to navigate to the URL localhost:4433 then I am getting  "localhost didn’t send any data. ERR_EMPTY_RESPONSE". I have attached the screenshot for the same. 
Note: This is my first experience working with docker, SSL, and Nginx.
Kindly Help me in this regard.

**Comment by baentsch** [comment]
Well, when you point a(ny) browser to this server, this browser('s software) needs to be enabled to use QSC cryptography -- and it looks like yours isn't. There are generally very few that try to already achieve QSC-interactions: Did you already take a look at https://github.com/open-quantum-safe/oqs-demos/tree/master/chromium ? For using QSC-TLS "application level interaction" I'd suggest using curl on the client side (see https://github.com/open-quantum-safe/oqs-demos/tree/master/curl )

**Comment by baentsch** [comment]
Fixed by #16
